rust/src/nfs/nfs.rs

```diff
@@ -438,6 +455,85 @@ impl NFSState {
         }
     }
 
+    fn add_rpc_udp_ts_frames(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], rpc_len: i64) -> (Option<Frame>, Option<Frame>) {
+        let rpc_udp_ts_pdu = Frame::new_ts(flow, stream_slice, input, rpc_len, NFSFrameType::RPCPdu as u8);
+        let rpc_udp_ts_creds = Frame::new_ts(flow, stream_slice, &input[24..], rpc_len - 32, NFSFrameType::RPCCreds as u8);
+        SCLogDebug!("RPC_UDP To Server Frames {:?} {:?}", rpc_udp_ts_pdu, rpc_udp_ts_creds);
+        (rpc_udp_ts_pdu, rpc_udp_ts_creds)
+    }
+
+    fn add_rpc_tcp_ts_frames(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], rpc_len: i64) -> (Option<Frame>, Option<Frame>) {
+        let rpc_tcp_ts_pdu = Frame::new_ts(flow, stream_slice, input, rpc_len, NFSFrameType::RPCPdu as u8);
+        let rpc_tcp_ts_creds = Frame::new_ts(flow, stream_slice, &input[28..], rpc_len - 36, NFSFrameType::RPCCreds as u8);
+        SCLogDebug!("RPC_TCP To Server Frames {:?} {:?}", rpc_tcp_ts_pdu, rpc_tcp_ts_creds);
+        (rpc_tcp_ts_pdu,rpc_tcp_ts_creds)
+    }
+
+    fn add_nfs_ts_frame(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], nfs_req_len: i64) -> Option<Frame> {
+        let nfs_req_pdu = Frame::new_ts(flow, stream_slice, input, nfs_req_len, NFSFrameType::NFSPdu as u8);
+        SCLogDebug!("NFS Request Frame {:?}", nfs_req_pdu);
+        nfs_req_pdu
+    }
+
+    fn add_nfs4_ts_pdu_frame(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], nfs4_req_len: i64) -> Option<Frame> {
+        let nfs4_ts_pdu = Frame::new_ts(flow, stream_slice, input, nfs4_req_len, NFSFrameType::NFS4Pdu as u8);
+        SCLogDebug!("NFS4 Request Frame: {:?}", nfs4_ts_pdu);
+        nfs4_ts_pdu
+    }
+
+    fn add_nfs4_ts_hdr_ops_frame(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], nfs4_req_len: i64) {
+        if nfs4_req_len > 0 {
+            let _nfs4_ts_hdr = Frame::new_ts(flow, stream_slice, input, 8, NFSFrameType::NFS4Hdr as u8);
+            let _nfs4_ts_ops = Frame::new_ts(flow, stream_slice, &input[8..], nfs4_req_len - 8, NFSFrameType::NFS4Ops as u8);
+            SCLogDebug!("NFS4 Hdr Frame {:?}", _nfs4_ts_hdr);
+            SCLogDebug!("NFS4 Ops Frame {:?}", _nfs4_ts_ops);
+        }
+    }
+
+    fn add_rpc_udp_tc_frames(&mut self, flow: *const Flow, stream_slice: &StreamSlice, input: &[u8], rpc_udp_len: i64) -> (Option<Frame>, Option<Frame>, Option<Frame>) {
```

Comment:
Q: why do we return the new frames in all functions? I do not see anything being done with them in the caller functions.

Reply:
I had a thought in mind for setting events for `RPC` frames for malformed data, but that still needs another look.
looks like the thought slipped to the other `nfs` frames while coding like a zombie xD

Can I leave the returned `RPC` frames for now? I'll remove the other ones they are inconsistent and not needed.
Thanks for pointing that out! :)

Reply:
You can leave anything. :P I was doing it myself and did not know if it is something we're supposed to do, and so, I asked ;)
Thanks for telling the use case :)